common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -274,30 +274,43 @@ def destroy(self, request, uuid):
             )
 
         if is_refund and entitlement.is_entitlement_refundable():
-            with transaction.atomic():
-                # Revoke and refund the entitlement
-                if entitlement.enrollment_course_run is not None:
-                    self._unenroll_entitlement(
-                        entitlement=entitlement,
-                        course_run_key=entitlement.enrollment_course_run.course_id,
-                        user=request.user
+            # Revoke the Course Entitlement and issue Refund
+            log.info(
+                'Entitlement Refund requested for Course Entitlement[%s]',
+                str(entitlement.uuid)
+            )
+            refund_status = refund_entitlement(course_entitlement=entitlement)
```

Comment:
Nit: since this returns a bool, it might be clearer to call this something like `refund_successful`.